morphir-ir/shared/src/main/scala/zio/morphir/ir/packages/Definition.scala

```diff
@@ -34,7 +38,46 @@ final case class Definition[+TA, +VA](
   def lookupTypeDefinition(moduleName: ModuleName): Option[ModuleDef[TA, VA]] =
     modules.get(moduleName).map(_.withPrivateAccess)
 
-  def mapDefinitionAttributes[TB, VB](tf: TA => VB, vf: VA => VB): Definition[TB, VB] = ???
+  def mapAttributes[TB, VB](tf: TA => TB, vf: VA => VB): Definition[TB, VB] = Definition(
+    modules.map { case (name, moduleDef) => (name, moduleDef.map(_.mapAttributes(tf, vf))) }
+  )
+
+  def lookupValueDefinition(path: Path, name: Name): Option[ValueDef[TA, VA]] =
+    self.lookupModuleDefinition(path).flatMap(_.lookupValueDefinition(name))
+
+//  @tailrec
+//  def selectModules(
+//      modulesToInclude: Set[ModuleName],
+//      name: PackageName
+//  ): Definition[TA, VA] = {
+//    val expandedMods = expandedModulesToInclude(modulesToInclude, name)
+//
+//    if (modulesToInclude == expandedMods) {
+//      Definition(modules.filter(module => modulesToInclude.contains(module._1)))
+//    } else selectModules(expandedMods, name)
+//  }
+//
+//  private def expandedModulesToInclude(modulesToInclude: Set[ModuleName], name: PackageName): Set[ModuleName] =
+//    findAllDependencies(modulesToInclude, name) ++ modulesToInclude
+//
+//  private def findAllDependencies(
+//      current: Set[ModuleName],
+//      name: PackageName
+//  ): Set[ModuleName] =
+//    current
+//      .foldLeft(Set.empty[ModuleName])((set, moduleName) =>
+//        modules
+//          .get(moduleName)
+//          .foldLeft(Set.empty[ModuleName]) { case (modDepSet, AccessControlled(_, modDef)) =>
```

Comment:
Let me know if theres something I could do about this line since we disabled Option2Iterable in scalafix

Reply:
Don't think I get how this is in play.